ampd/src/tofnd/grpc.rs

```diff
@@ -0,0 +1,62 @@
+use async_trait::async_trait;
+use error_stack::{IntoReport, Result};
+use mockall::automock;
+use tonic::transport::Channel;
+use tonic::{Response, Status};
+
+use super::proto::{
+    key_presence_response::Response as KeyPresenceEnum, keygen_response::KeygenResponse,
+    multisig_client, sign_response::SignResponse, KeyPresenceRequest, KeygenRequest, SignRequest,
+};
+
+#[automock]
+#[async_trait]
+pub trait MultisigClient {
+    async fn keygen(&mut self, request: KeygenRequest) -> Result<KeygenResponse, Status>;
+    async fn sign(&mut self, request: SignRequest) -> Result<SignResponse, Status>;
+    async fn key_presence(
+        &mut self,
+        request: KeyPresenceRequest,
+    ) -> Result<KeyPresenceEnum, Status>;
+}
+
+#[async_trait]
+impl MultisigClient for multisig_client::MultisigClient<Channel> {
```

Comment:
Add a `connect` method that wraps `multisig_client::MultisigClient::connect` and returns `Self`.